src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -43,63 +46,207 @@ namespace intel_cpu {
+        attrs.outMemPrecs.size() != rhs.attrs.outMemPrecs.size())
+        return false;
+
+    bool retVal = true;
+    for (size_t i = 0; i < attrs.inMemBlockedDims.size(); i++)
+        retVal = retVal && attrs.inMemBlockedDims[i] == rhs.attrs.inMemBlockedDims[i];
+    for (size_t i = 0; i < attrs.inMemOrders.size(); i++)
+        retVal = retVal && attrs.inMemOrders[i] == rhs.attrs.inMemOrders[i];
+    for (size_t i = 0; i < attrs.inMemPrecs.size(); i++)
+        retVal = retVal && attrs.inMemPrecs[i] == rhs.attrs.inMemPrecs[i];
+    for (size_t i = 0; i < attrs.outMemBlockedDims.size(); i++)
+        retVal = retVal && attrs.outMemBlockedDims[i] == rhs.attrs.outMemBlockedDims[i];
+    for (size_t i = 0; i < attrs.outMemOrders.size(); i++)
+        retVal = retVal && attrs.outMemOrders[i] == rhs.attrs.outMemOrders[i];
+    for (size_t i = 0; i < attrs.outMemPrecs.size(); i++)
+        retVal = retVal && attrs.outMemPrecs[i] == rhs.attrs.outMemPrecs[i];
+
+    return retVal;
+}
+
+snippets::op::Subgraph::BlockedShapeVector getBlockedShapes(const std::vector<std::vector<size_t>>& memBlockedDims,
+        const std::vector<std::vector<size_t>>& memOrders, const std::vector<InferenceEngine::Precision>& memPrecs) {
+    size_t numShapes = memBlockedDims.size();
+    if (memOrders.size() != numShapes || memPrecs.size() != numShapes)
+        IE_THROW(Unexpected) << "Number of shapes is mismacthed for dimensions, orders and precisions";
+    snippets::op::Subgraph::BlockedShapeVector blockedShapes(numShapes);
+    for (size_t i = 0; i < numShapes; i++) {
+        size_t dimSize = memBlockedDims[i].size();
+        std::vector<Dimension> dims(dimSize);
+        for (size_t j = 0; j < dimSize; j++) {
+            dims[j] = memBlockedDims[i][j];
+        }
+        ov::PartialShape shape(dims);
+        ov::AxisVector order(memOrders[i]);
+        ov::element::Type precision = InferenceEngine::details::convertPrecision(memPrecs[i]);
+
+        blockedShapes[i] = snippets::op::Subgraph::BlockedShape{shape, order, precision};
+    }
+
+    return blockedShapes;
+}
+
 class SnippetShapeInfer : public ShapeInferEmptyPads {
 public:
-    SnippetShapeInfer(Snippet* node) : m_node(node) {}
+    SnippetShapeInfer(std::shared_ptr<ov::Model> body) : m_body(body) {}
     Result infer(
         const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
         const std::unordered_map<size_t, MemoryPtr>& data_dependency) override {
-        return {m_node->shapeInfer(), ShapeInferStatus::success};
+        auto broadcast_merge = [](VectorDims& dst, const VectorDims& src) {
+            // Ranks are both static.
+            auto dst_rank = dst.size();
+            auto src_rank = src.size();
+            const auto new_rank = std::max(dst_rank, src_rank);
+            dst.insert(dst.begin(), new_rank - dst_rank, 1);
+            bool success = true;
+            for (size_t i = 0; i < new_rank; i++) {
```

Comment:
```suggestion
            for (size_t i = 0; i < new_rank && success; i++) {
```

Reply:
Here success means the dst dimenion is updated, not a fail of broadcast. Need continue broadcast for next dimension. seems success here is not needed and removed. welcome comments if concerns. thanks Alexandra!

Reply:
Should we throw exception anyway if dimensions aren't broadcastable? What do you think?

Reply:
I agree, added, thanks Alexandra!